backend/companies/tests/test_company_api.py

```diff
@@ -0,0 +1,95 @@
+from unittest import mock
+
+import pytest
+from django.test import override_settings
+from requests.exceptions import HTTPError
+
+from companies.api.v1.serializers import CompanySerializer
+from companies.models import Company
+from companies.tests.data.company_data import (
+    DUMMY_COMPANY_DATA,
+    DUMMY_YTJ_BUSINESS_DETAILS_RESPONSE,
+    DUMMY_YTJ_RESPONSE,
+)
+
+
+def get_company_api_url():
+    return "/v1/company/"
+
+
+@pytest.mark.django_db
+@override_settings(MOCK_FLAG=True)
+def test_get_mock_company(api_client):
+    response = api_client.get(get_company_api_url())
+
+    assert response.status_code == 200
+
+    company = Company.objects.first()
+    company_data = CompanySerializer(company).data
+
+    assert response.data == company_data
+    assert response.data["business_id"] == DUMMY_COMPANY_DATA["business_id"]
+
+
+@pytest.mark.django_db
+@override_settings(MOCK_FLAG=True)
+def test_get_mock_company_results_in_error(api_client):
+    api_client.credentials(HTTP_SESSION_ID="-1")
+    response = api_client.get(get_company_api_url())
+
+    assert response.status_code == 500
+    assert (
+        response.data
+        == "YTJ API is under heavy load or no company found with the given business id"
+    )
+
+
+@pytest.mark.django_db
+@override_settings(MOCK_FLAG=False)
+def test_get_company_from_ytj(api_client):
+    with mock.patch(
+        "companies.ytj.ytj_client.YTJClient._get",
+        side_effect=[DUMMY_YTJ_RESPONSE, DUMMY_YTJ_BUSINESS_DETAILS_RESPONSE],
+    ):
+        response = api_client.get(get_company_api_url())
+
+    assert response.status_code == 200
+
+    company = Company.objects.first()
+    company_data = CompanySerializer(company).data
+
+    assert response.data == company_data
+    assert (
+        response.data["business_id"] == DUMMY_YTJ_RESPONSE["results"][0]["businessId"]
+    )
+
+
+@pytest.mark.django_db
+@override_settings(MOCK_FLAG=False)
+def test_get_company_from_ytj_results_in_error(api_client):
+    with mock.patch(
```

Comment:
You could optionally also use something like (requests-mock)[https://pypi.org/project/requests-mock/] for mocking the responses.